geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/index/state/ExtentsEventTimeExpiryOrderingFeatureStateFactory.scala

```diff
@@ -0,0 +1,47 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.kafka.index.state
+
+import java.util.concurrent.ScheduledExecutorService
+
+import org.locationtech.geomesa.kafka.index.state.FeatureStateFactory.{ExpiredState, ExtentsExpiryState, FeatureExpiration, FeatureState}
+import org.locationtech.geomesa.utils.cache.Ticker
+import org.locationtech.geomesa.utils.index.SpatialIndex
+import org.opengis.feature.simple.SimpleFeature
+import org.opengis.filter.expression.Expression
+
+/**
+  * Non-point feature index with event time ordering and timed expiration
+  *
+  * @param index spatial index
+  * @param geom geometry attribute
+  * @param executor executor
+  * @param expiry expiry in millis
+  * @param eventTime event time expression
+  */
+class ExtentsEventTimeExpiryOrderingFeatureStateFactory(index: SpatialIndex[SimpleFeature],
```

Comment:
*blinks*  Ok, so I'm reading quickly... why do some of these have an Executor and others not?

Reply:
the ones that expire features have an executor